docs/src/modules/components/ExamplesGrid/ExamplesGrid.tsx

```diff
@@ -44,15 +44,11 @@ function CodeSandboxIcon() {
 }
 
 function Templates({ examplesFile }: TemplatesProps) {
-  const [examples, setExamples] = React.useState<Example[]>([]);
-
-  React.useEffect(() => {
-    const importExamples = async () => {
-      const exampleContent = await import(`./${examplesFile}`);
-      setExamples(exampleContent.default);
-    };
-    importExamples();
-  }, [examplesFile]);
+  // @ts-ignore
+  const req = require.context('./', false);
```

Comment:
I'd just import them statically instead of doing proprietary webpack stuff.

```tsx
import coreExamples from './core-examples'
import studioExamples from './studio-examples'

const EXAMPLES_BY_PRODUCT = {
  'core-examples': coreExamples,
  'studio-examples': studioExamples
}

// ...

const examples = EXAMPLES_BY_PRODUCT[examplesFile]
```

Or better yet, just pass the array as a prop and import thecorrect file on the callsite.

Reply:
Yeah, true, this indirection is strange. I have created one file per demo, it should have been done this way in the first place IMHO.